dash/include/dash/halo/HaloMatrixWrapper.h

```diff
@@ -267,8 +264,10 @@ class HaloMatrixWrapper {
    * Initiates a blocking halo region update for all halo elements.
    */
   void update() {
-    for(auto& region : _region_data)
-      update_halo_intern(region.second, false);
+    for(auto& region : _region_data) {
+      update_halo_intern(region.second);
+      dart_wait_local(&region.second.handle);
```

Comment:
I suggest moving the wait into a separate loop to better overlap multiple transfers. If possible, using `dart_waitall_local` would be more efficient.